.github/workflows/release.yaml

```diff
@@ -24,6 +24,19 @@ jobs:
           fetch-depth: 0
       - name: Checkout - force fetch tags
         run: git fetch --force --tags
+      - name: Tagged build - generate tag
+        if: startsWith(github.ref, 'refs/tags/')
+        run: |
+          echo "CURRENT_TAG=$GITHUB_REF_NAME" >> $GITHUB_ENV
+      - name: Snapshot build - generate tag
+        if: ${{ !startsWith(github.ref, 'refs/tags/') }}
+        run: |
+          sudo apt-get install -y --no-install-recommends python3-semver
+          export NEXT_TAG=$(pysemver nextver $(git describe --abbrev=0) patch)
```

Comment:
Maybe a possible small improvement here for the 1st build when there are no annotated tags available yet is to default to `0.0.0` lets say.

```shell
export NEXT_TAG=$(pysemver nextver $(git describe --abbrev=0) patch || echo "0.0.0")
```